docs/source/examples/sentiment.rst

```diff
@@ -0,0 +1,359 @@
+
+However, using the stack, Recursive Neural Network can perform mini batch parallel calculation.
+
+3.1 Preparation of Dataset, Iterator
+-------------------------------------
+
+First, we convert the recursive calculation of Recursive Neural Network to a serial calculation using a stack.
+For each node of the tree structure dataset, numbers are assigned to each node in "returning order" as follows.
+
+.. figure:: ../../image/sentiment/returning-order.png
+    :scale: 50%
+
+The returning order is a procedure of numbering nodes of a tree structure. It is a procedure of attaching
+a smaller number to all child nodes than the parent node. If we process nodes in descending order of numbers,
+we can trace the child nodes before the parent node.
+
+.. literalinclude:: ../../../examples/sentiment/train_recursive_minibatch.py
+   :language: python
+   :pyobject: linearize_tree
+   :caption: train_recursive_minibatch.py
+
+.. literalinclude:: ../../../examples/sentiment/train_recursive_minibatch.py
+   :language: python
+   :start-after: parser.parse_args()
+   :end-before: model = ThinStackRecursiveNet
+   :caption: train_recursive_minibatch.py
+   :dedent: 4
+
+3.2 Definition of mini-batchable models
+----------------------------------------
+
+Recursive Neural Network has two operations: Operation A for computing an embedding vector for the leaf node.
+Operation B for computing the hidden state vector of the parent node from the hidden state vectors of the two
+child nodes.
+
+For each sample, we assign index to each node in returning order. If we traverse the node in return order,
+we will find that operation A is performed on the leaf node and operation B is performed at the other nodes.
+
+This operation can also be regarded as using a stack to scan a tree structure. A stack is a last-in,
+x first-out data structure that allows we to do two things: a push operation to add data and a pop operation
+to get the last pushed data.
+For operation A, push the calculation result to the stack. For operation B, pop two data and push the new
+calculation result.
+
+When we parallelize the above operation, it is necessary to traverse nodes and perform operation A and operation
+B precisely because the tree structure is different for each sample. However, by using the stack,
+we can calculate different tree structures by simple repeating processing. Therefore, parallelization is possible.
+
+.. literalinclude:: ../../../examples/sentiment/thin_stack.py
+   :language: python
+   :pyobject: ThinStackSet
+   :caption: thin_stack.py
+
+Let the sentence length be :math:`I` and the number of dimensions of the hidden vector be :math:`D`,
+the thin stack can efficiently use the memory by using the matrix of :math:`(2I-1) \times D`.
+In a normal stack, we need :math:`O(I^2 D)` space computation, whereas thin stacks require :math:`O(ID)`.
+It is realized by push operation ``thin_stack_set`` and pop operation ``thin_stack_get``.
```

Comment:
This is realized by push operations ``thin_stack_set`` and pop operations ``thin_stack_get``.